src/main/resources/assets/admin/common/js/form/set/FormSetOccurrenceView.ts

```diff
@@ -23,20 +26,63 @@ module api.form {
 
         protected occurrenceContainerClassName: string;
 
+        private dirtyFormItemViewsMap: object = {};
+
+        private deleteOccurrenceConfirmationDialog: ConfirmationDialog;
+
+        private formDataChangedListener: (event: PropertyValueChangedEvent) => void;
+
         constructor(className: string, formItemOccurrence: FormItemOccurrence<FormItemOccurrenceView>) {
             super(className, formItemOccurrence);
+
+            this.initConfirmationDialog();
+            this.initFormDataChangeListener();
+        }
+
+        private initConfirmationDialog() {
+            this.deleteOccurrenceConfirmationDialog = new ConfirmationDialog()
+                .setQuestion(i18n('dialog.confirm.occurrences.delete'))
+                .setYesCallback(() => {
+                    this.notifyRemoveButtonClicked();
+                });
+        }
+
+        private initFormDataChangeListener() {
+            this.formDataChangedListener = (event: PropertyValueChangedEvent) => {
+                this.formItemViews.filter(
+                    (formItemView: FormItemView) => formItemView.getFormItem().getName() === event.getProperty().getName()).forEach(
```

Comment:
Move ``event.getProperty().getName()`` and ``event.getNewValue()`` up, out of ``filter``/``forEach`` loops and assign to const - no point in fetching these values in each iteration. Same with ``formItemView.getFormItem().getName()`` - it's called 4 times inside the ``forEach`` loop.